### DISCUSSION TITLE
Using ollama to interact based on all notes in vault?

### DISCUSSION THREAD

**Discussion by tonzyl** [pfrankov/obsidian-local-gpt#5] [General]
Hi Pavel, thank you for this plugin. I came across this blogpost by ollama https://ollama.ai/blog/llms-in-obsidian where they describe indexing an entire vault and then using ollama to interact based on all notes. From your description and actions, am I correct in assuming your plugin does not do such indexing? Would this be something you want to add?

**Comment by pfrankov** [comment]
Hi there! Yes, that's correct — there is no ability to ask your notes but it exists in my plans. I'm not sure that it's really that simple as it is described in the blogpost but at least I'll try to implement it somewhere in the future.

**Comment by pfrankov** [comment]
@tonzyl by the way, could you also provide some use cases for interacting with your notes?

**Reply by tonzyl** [reply]
I realise I never answered your earlier question. My use case would be to ask the bot questions and have it react from my full vault as input. A conversation with my self (as far as reflected in my notes). To spot associations that I can then make explicit in my notes. A chatbot as mirror, that surfaces related content.

**Reply by pfrankov** [reply]
@tonzyl thanks! It looks like there are some Obsidian plugins to chat with your notes. For example https://github.com/longy2k/obsidian-bmo-chatbot
LocalGPT plugin in the other hand is not for chatting, its main purpose is to help you with writing, with text editing, with different types of summarization.
In your opinion, are there any scenarios where local context of the vault may be helpful in text editing?

**Comment by tonzyl** [comment]
Yes, I can see two, from my personal practice. 
My notes are my own, text editing might better reflect my own writing style / vocabulary if it is based on what's in my vault (as it covers 'my' topics, and my style). This can be helpful when writing within the vault (so my future self understands my current notes better), or for external purposes (combining notes into a new one that will be a blogpost or essay) (Of course this doesn't apply to the excerpts and quotes if that is the focus of a vault.)
The second one is maybe more an edge-case: my notes are not monolingual (English and Dutch mostly wrritten by me, but also contains German texts by me, and  a few source texts in other languages like French, Spanish, Italian, Russian). Pulling from that linguistic variety of sources to bring it together in a coherent single language text (for output e.g.) might be a good usage of AI's editing capabilities.
